spec/index.bs

```diff
@@ -6682,6 +6726,58 @@ interface mixin GPUProgrammablePassEncoder {
     Otherwise return `true`.
 </div>
 
+<div algorithm>
+    <dfn abstract-op>Encoder bind groups alias a writable resource</dfn>(|encoder|, |pipeline|)
+
+    **Arguments:**
+    <dl>
+        : {{GPUProgrammablePassEncoder}} |encoder|
+        :: Encoder whose bind groups are being validated.
+        : {{GPUPipelineBase}} |pipeline|
+        :: Pipeline to validate |encoder|s bind groups are compatible with.
+    </dl>
+
+    1. For each |stage| in [{{GPUShaderStage/VERTEX}}, {{GPUShaderStage/FRAGMENT}}, {{GPUShaderStage/COMPUTE}}]:
+        1. Let |writableBufferBindings| be an [=ordered set=]&lt;{{GPUBufferBinding}}&gt;.
+        1. Let |writableTextureViews| be an [=ordered set=]&lt;{{GPUTextureView}}&gt;.
+        1. For each pair of ({{GPUIndex32}} |index|, {{GPUBindGroupLayout}} |bindGroupLayout|) in
+            |pipeline|.{{GPUPipelineBase/[[layout]]}}.{{GPUPipelineLayout/[[bindGroupLayouts]]}}:
+            1. Let |bindGroupEntries| be
+                |encoder|.{{GPUProgrammablePassEncoder/[[bind_groups]]}}[|index|].{{GPUBindGroupDescriptor/entries}}.
+            1. Let |bindGroupLayoutEntries| be
+                |bindGroupLayout|.{{GPUBindGroupLayout/[[descriptor]]}}.{{GPUBindGroupLayoutDescriptor/entries}}.
+            1. For each {{GPUBindGroupEntry}} |bindGroupLayoutEntry| in |bindGroupLayoutEntries|
+                for which |bindGroupLayoutEntry|.{{GPUBindGroupLayoutEntry/visibility}} contains |stage|:
+                1. Let |bindGroupEntry| be the {{GPUBindGroupEntry}} in |bindGroupEntries| for which
+                    |bindGroupEntry|.{{GPUBindGroupEntry/binding}} is equal to
+                    |bindGroupLayoutEntry|.{{GPUBindGroupLayoutEntry/binding}}.
+                1. If |bindGroupEntry|.{{GPUBindGroupEntry/resource}} is a {{GPUBufferBinding}}:
+
+                    1. Let {{GPUBufferBinding}} |bufferBinding| be |bindGroupEntry|.{{GPUBindGroupEntry/resource}}.
+                    1. For each {{GPUBufferBinding}} |existingWritableBufferBinding| in |writableBufferBindings|,
+                        if |existingWritableBufferBinding| and |bufferBinding| are [=buffer-binding-aliasing=], return `true`.
```

Comment:
Sorry to bring this up, but we may need to come back to the discussion table and clarify if aliasing between VS and FS is still considered a hazard. It's not clear because those are separate programs. And we already have a situation where separate programs from separate pipelines may be using the same writable resources (within a render pass), so this isn't too far from it.

Reply:
You're right, I only was trying to resolve a comment on the other PR but didn't think about this hard enough.

I am pretty sure vertex shaders aren't all guaranteed to complete before fragment shaders start. If that's correct then I should just remove the "for each stage" part I think.